src/ucs/datastruct/callbackq.h

```diff
@@ -152,6 +152,7 @@ void ucs_callbackq_remove_safe(ucs_callbackq_t *cbq, int id);
 
 /**
  * Dispatch callbacks from the callback queue.
+ * Must be called from single thread only.
```

Comment:
should we have it enforced in the code somehow? with a lock or something

Reply:
the user is responsible to use a lock if needed, if we add it here it would be extra overhead for single-thread use case